ptvsd/wrapper.py

```diff
@@ -1580,17 +1585,34 @@ def on_setBreakpoints(self, request, args):
                 self.bp_map.remove(pyd_bpid, vsc_bpid)
 
         cmd = pydevd_comm.CMD_SET_BREAK
-        msgfmt = '{}\t{}\t{}\t{}\tNone\t{}\tNone\t{}'
+        msgfmt = '{}\t{}\t{}\t{}\tNone\t{}\t{}\t{}\t{}'
         for src_bp in src_bps:
             line = src_bp['line']
             vsc_bpid = self.bp_map.add(
                     lambda vsc_bpid: (path, vsc_bpid))
             self.path_casing.track_file_path_case(path)
+
             hit_condition = self._get_hit_condition_expression(
                                 src_bp.get('hitCondition', None))
-            msg = msgfmt.format(vsc_bpid, bp_type, path, line,
-                                src_bp.get('condition', None),
-                                hit_condition)
+            logMessage = src_bp.get('logMessage', '')
+            if len(logMessage) == 0:
+                is_logpoint = None
+                condition = src_bp.get('condition', None)
+                expression = None
+            else:
+                is_logpoint = True
+                condition = None
+                expressions = re.findall('\{.*?\}', logMessage)
+                if len(expressions) == 0:
+                    expression = 'print("{}")'.format(logMessage.replace('"', '\\"')) # noqa
```

Comment:
This also needs to escape backslashes. But I think the easiest way to do it here is to use repr, and let Python do the escaping and add the quotes: `print({}).format(repr(logMessage))`

Reply:
:+1: